### ISSUE TITLE
LDC outputs LD warnings on every compile

### ISSUE THREAD

**Issue by JackStouffer** [ldc-developers/ldc#1235]
Using the homebrew version of ldc2 0.16.0, compiling any file outputs a host of warnings from ld. Example:

```
void main() {}
```

```
$ ldc2 test.d
ld: warning: object file (test.o) was built for newer OSX version (15.0) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(sections_ldc.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(dmain2.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(ti_long.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(ti_Ag.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(object_.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(common.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(minfo.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(memory.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(errno.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(critical_.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(monitor_.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(adi.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(cast_.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(libunwind.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(proxy.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(lifetime.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(runtime.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(thread.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(ti_char.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(ti_ubyte.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(ti_byte.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(ti_ulong.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(ti_wchar.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(ti_void.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(ti_dchar.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(gc.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(stats.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(utf.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(hash.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(string.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(tlsgc.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(config.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(common.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(mutex.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(time.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(bitop.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(memory.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(demangle.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(aaA.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(qsort.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(arrayinit.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(exception.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(ti_float.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(ti_int.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(ti_uint.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(ti_Ashort.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(ti_Aint.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(os.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(bits.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(config.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(random.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(exception.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(aApply.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(ti_short.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(ti_ushort.o)) was built for newer OSX version (14.5) than being linked (10.11)
ld: warning: object file (/usr/local/Cellar/ldc/0.16.1/lib/libdruntime-ldc.a(switch_.o)) was built for newer OSX version (14.5) than being linked (10.11)
```

**Comment by kinke** [comment]
I think that's fixed in master already.

**Comment by smolt** [comment]
Yes, master fixes it.

Related, LDC will eventually need option for -mmacosx-version-min to specify what version of OS X the binaries it produces are compatible with.  It is being enforced by linker for watchOS and soon for iOS, tvOS, and probably OS X.

**Comment by smolt** [comment]
Fixed by #1209
